core/dbt/contracts/project.py

```diff
@@ -180,6 +180,13 @@ class RegistryPackageMetadata(
 }
 
 
+@dataclass
+class SchemaManagementConfiguration(HyphenatedDbtClassMixin, Replaceable):
+    database: Optional[str] = None
+    schema: Optional[str] = None
+    action: Optional[str] = None
```

Comment:
Confirming that the current options for this are "warn" and "drop"? We probably want to perform that validation here (`Enum` type, or custom `validate` method), rather than accidentally find we have an invalid value only at the very end of runtime

Reply:
warn and drop are the 2 options that do something indeed, as seen [here](https://github.com/BigDataRepublic/dbt-core/blob/feature/ct-426-manage-schemas/core/dbt/task/run.py#L534). Any other option will do disable this feature. So right now, typos will likely disable the table dropping rather than accidentally enabling it.

If we turn the field into an enum-like type, I think we should have 3 options:
- warn
- drop
- none / noop / disabled / ignore / skip